src/NServiceBus.AwsLambda.SQS.Analyzer/AwsLambdaDiagnostics.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace NServiceBus.AwsLambda.SQS.Analyzer
+{
+    using System.Security.Cryptography;
+    using Microsoft.CodeAnalysis;
+
+    public static class AwsLambdaDiagnostics
+    {
+        public const string PurgeOnStartupNotAllowedId = "NSBLAM001";
+        public const string LimitMessageProcessingToNotAllowedId = "NSBLAM002";
+        public const string DefineCriticalErrorActionNotAllowedId = "NSBLAM003";
+        public const string SetDiagnosticsPathNotAllowedId = "NSBLAM004";
+        public const string MakeInstanceUniquelyAddressableNotAllowedId = "NSBLAM005";
+        public const string UseTransportNotAllowedId = "NSBLAM006";
+        public const string OverrideLocalAddressNotAllowedId = "NSBLAM007";
+        public const string RouteReplyToThisInstanceNotAllowedId = "NSBLAM008";
+        public const string RouteToThisInstanceNotAllowedId = "NSBLAM009";
+        public const string TransportTransactionModeNotAllowedId = "NSBLAM010";
+
+        const string DiagnosticCategory = "NServiceBus.AwsLambda";
+
+        internal static readonly DiagnosticDescriptor PurgeOnStartupNotAllowed = new DiagnosticDescriptor(
+             id: PurgeOnStartupNotAllowedId,
+             title: "PurgeOnStartup is not supported in Aws Lambda",
```

Comment:
Need to check if this should be AWS Lambda

Reply:
It is AWS in all our docs so I've changed it.